guides/release/routing/specifying-a-routes-model.md

```diff
@@ -9,118 +11,128 @@ Router.map(function() {
+    return this.get('ajax').request('https://some-api-endpoint.com/posts', {
+      method: 'GET'
+    });
   }
 });
 ```
 
-In the `model` hook for routes with dynamic segments, it's your job to
-turn the ID (something like `47` or `post-slug`) into a model that can
-be rendered by the route's template. In the above example, we use the
-photo's ID (`params.photo_id`) as an argument to Ember Data's `findRecord`
-method.
-
-Note: A route with a dynamic segment will always have its `model` hook called when it is entered via the URL.
-If the route is entered through a transition (e.g. when using the [link-to](../../templates/links/) Handlebars helper),
-and a model context is provided (second argument to `link-to`), then the hook is not executed.
-If an identifier (such as an id or slug) is provided instead then the model hook will be executed.
-
-For example, transitioning to the `photo` route this way won't cause the `model` hook to be executed (because `link-to`
-was passed a model/):
-
-```handlebars {data-filename=app/templates/photos.hbs}
-<h1>Photos</h1>
-{{#each this.model as |photo|}}
-  <p>
-    {{#link-to "photo" photo}}
-      <img src="{{photo.thumbnailUrl}}" alt="{{photo.title}}" />
-    {{/link-to}}
-  </p>
-{{/each}}
+Lastly, you could hand-write a promise that you return from the model hook:
+
+```javascript {data-filename=app/routes/favorite-posts.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return new Promise((resolve, reject) => {
+      // get some data
+      resolve( /* data goes here */);
+    })
+  }
+});
 ```
 
-while transitioning this way will cause the `model` hook to be executed (because `link-to` was passed `photo.id`, an
-identifier, instead):
+Just like in the static data example, the results of an asynchronous request in the `model` hook can be used in the template:
 
-```handlebars {data-filename=app/templates/photos.hbs}
-<h1>Photos</h1>
-{{#each this.model as |photo|}}
-  <p>
-    {{#link-to "photo" photo.id}}
-      <img src="{{photo.thumbnailUrl}}" alt="{{photo.title}}" />
-    {{/link-to}}
-  </p>
+```handlebars {data-filename=app/templates/favorite-posts.hbs}
+{{#each this.model as |post|}}
+  <div>
+    {{post.title}}
+  </div>
 {{/each}}
 ```
 
-Routes without dynamic segments will always execute the model hook.
+Behind the scenes, what is happening is that the route's controller receives the results of the model hook, and makes those results available to the template. Your app may not have a controller file for the route, but the behavior is the same regardless.
 
 ## Multiple Models
 
+What should you do if you need the `model` to return the results of multiple API requests?
+
 Multiple models can be returned through an
 [RSVP.hash](https://www.emberjs.com/api/ember/release/classes/rsvp/methods/hash?anchor=hash).
 The `RSVP.hash` method takes an object with promises or values as properties as an argument, and returns a single promise.
```

Comment:
I think this might be a typo: " as properties as an argument"? I don't quite understand what this sentence means... 

Based on the [API docs](https://api.emberjs.com/ember/release/classes/rsvp/methods/hash?anchor=hash), it sounds like the `hash` method takes an object (which can contain multiple promises), and returns a promise. The returned promise will resolve when all the promises listed IN the object are resolved, or rejected if ANY of them are rejected.

With the code example below, it may be helpful to show in comments what the hash will resolve to? Does `songs` resolves to an array of song records?

Reply:
I reworded this, based on the API docs phrasing.